src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -31,39 +44,47 @@ public String todayDate(DateTimePart datePart) {
      *                   - 3-й элемент массива - день (число);
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        StringBuilder date = new StringBuilder();
+        date = dateParams.length == 0 ? date
+                : date.append(dateParams[0]).append(dateParams[1]).append(dateParams[2]);
+        try {
+            LocalDate localDate = LocalDate.parse(date, FORMATTER1);
+            return Optional.of(localDate);
+        } catch (DateTimeParseException e) {
+            return Optional.empty();
```

Comment:
move it out of catch block, we should not use try catch as if else

Reply:
done